online/src/main/scala/ai/chronon/online/CatalystUtil.scala

```diff
@@ -191,11 +190,30 @@ class CatalystUtil(expressions: collection.Seq[(String, String)],
         projectFunc
       }
       case ProjectExec(projectList, childPlan) => {
-        val unsafeProjection = UnsafeProjection.create(projectList, childPlan.output)
-        def projectFunc(row: InternalRow): Option[InternalRow] = {
-          Some(unsafeProjection.apply(row))
+        childPlan match {
```

Comment:
should we add a comment here indicating that this is subtly different from the prior codegen block?